src/java.base/share/classes/java/lang/invoke/GenerateJLIClassesHelper.java

```diff
@@ -53,16 +55,23 @@
     private static final String SPECIES_RESOLVE = "[SPECIES_RESOLVE]";
 
     static void traceLambdaForm(String name, MethodType type, Class<?> holder, MemberName resolvedMember) {
-        if (TRACE_RESOLVE) {
-            System.out.println(LF_RESOLVE + " " + holder.getName() + " " + name + " " +
-                    shortenSignature(basicTypeSignature(type)) +
-                    (resolvedMember != null ? " (success)" : " (fail)"));
+        if (TRACE_RESOLVE || CDS.isDumpLoadedClassList()) {
+            String traceLF = LF_RESOLVE + " " + holder.getName() + " " + name + " " +
+                    shortenSignature(basicTypeSignature(type));
+            if (TRACE_RESOLVE) {
+                System.out.println(traceLF + (resolvedMember != null ? " (success)" : " (fail)"));
+            }
+            CDS.traceLambdaFormInvoker(traceLF);
         }
     }
 
     static void traceSpeciesType(String cn, Class<?> salvage) {
-        if (TRACE_RESOLVE) {
-            System.out.println(SPECIES_RESOLVE + " " + cn + (salvage != null ? " (salvaged)" : " (generated)"));
+        if (TRACE_RESOLVE || CDS.isDumpLoadedClassList()) {
+            String traceSP = SPECIES_RESOLVE + " " + cn;
+            if (TRACE_RESOLVE) {
+                System.out.println(traceSP + (salvage != null ? " (salvaged)" : " (generated)"));
+            }
+            CDS.traceLambdaFormInvoker(traceSP);
```

Comment:
I suggest leaving the existing code unchanged.  Instead, add the following:
```
    if (CDS.isDumpingClassList()) {
         CDS.traceSpeciesType(cn);
    }
```

The above uses Ioi's suggested method name which reads better.

Reply:
Done